qemu/tests/rh_qemu_update.py

```diff
@@ -0,0 +1,261 @@
+                    host_qemu["nvr"],
+                    guest_qemu["nvr"]
+                )
+            )
+        else:
+            test.cancel(
+                "Available qemu-kvm '{}' is older or same compared to guest's '{}'".format(
+                    host_qemu["nvr"],
+                    guest_qemu["nvr"]
+                )
+            )
+        if "module_id" in host_qemu:
+            return host_qemu['module_id']
+        else:
+            return host_qemu['nvr']
+
+    def update_qemu(session, install_id):
+        """
+        Prepare repository containing the newest version of qemu-kvm package,
+        handle modules if needed and run upgrade to the newest version
+        """
+        def nice_format(ugly_format):
+            better_format = ugly_format.replace("  ", " ")
+            if "  " in better_format:
+                return nice_format(better_format)
+            else:
+                return better_format
+        build_or_module = "cm_cmd_brew"
+        if install_id.isnumeric():
+            build_or_module = "cm_cmd_module"
+        # Prepare module or build repo containing newer version of qemu-kvm
+        cmd = f"python3 {params['cm_path']}{params[build_or_module]} {install_id}"
+        test.log.info(f"Running: {nice_format(cmd)}")
+        try:
+            session.cmd(cmd, timeout=OPERATION_TIMEOUT)
+            test.log.info("Creation of repo '{}' succeeded".format(
+                install_id
+                )
+            )
+        except Exception as e:
+            test.error("Creation of repo '{}' failed with: {}".format(
+                install_id, e
+                )
+            )
+        # Disable and enable new module if module is used
+        if install_id.isnumeric():
+            stream = _get_module_stream(session, install_id)
+            disable_cmd = f"{params['pm_tool']} module --disable virt:{stream}"
+            s_disable, o_disable = session.cmd_status_output(
+                disable_cmd, timeout=OPERATION_TIMEOUT
+                )
+            if s_disable != 0:
+                test.fail("Disable of module virt failed with: {}".format(
+                    o_disable
+                    )
+                )
+            else:
+                test.log.info("Disable of module virt succeeded")
+            enable_cmd = f"{params['pm_tool']} module --enable virt:{stream}"
+            s_enable, o_enable = session.cmd_status_output(
+                enable_cmd, timeout=OPERATION_TIMEOUT
+                )
+            if s_enable != 0:
+                test.fail("Enable of module virt:{} failed with: {}".format(
+                    stream, o_enable
+                    )
+                )
+            else:
+                test.log.info("Enable of module virt:{} succeeded".format(
+                    stream
+                    )
+                )
+        # Run upgrade to newer qemu-kvm version
+        cmd_upgrade = "{} upgrade -y {}".format(
```

Comment:
@yanan-fu I see your point. I run `upgrade` only for qemu-kvm package in both cases (brew build, module). So I in case of module I enable the newer module and run upgrade of single package from this module. Do you think it is invalid, please? I consider this whole test case to be something focused to package upgrade only at the end.

Reply:
Yes, it can work but it is not the recommend usage for RHEL8 product.  Those components in a virt module should work together. Or we can say, module upgrade fail is a product issue too.

Reply:
This is good argument, let's follow recommended way. I will adapt the code it will be small change :)

Reply:
Change pushed.